ci-operator/step-registry/openshift/e2e/openstack/perfscale/network-perf/openshift-e2e-openstack-perfscale-network-perf-workflow.yaml

```diff
@@ -0,0 +1,17 @@
+workflow:
+  as: openshift-e2e-openstack-perfscale-network-perf
+  steps:
+    pre:
+    - ref: openstack-provision-securitygroups-netperf
+    - chain: ipi-openstack-pre
```

Comment:
The problem you have now is that:
* `openstack-provision-securitygroups-netperf` requires clouds.yaml, that is generated by a step contained in the `openstack-provision-securitygroups` chain;
* but `openstack-conf-installconfig`, which is contained in the `ipi-openstack-pre` chain, must run after `openstack-provision-securitygroups-netperf`.

I suppose that there are multiple solutions, but here's one:
1. create a new step (replacing `openstack-provision-securitygroups-netperf`) called `openstack-provision-securitygroups`
2. in `openstack-provision-securitygroups`, you look at one environment variable (like `ADDITIONAL_SECURITY_GROUP_RULES` for example). If that environment variable contains `netperf`, then you create a security group with the `netperf` rule and you add its ID to the file in `SHARED_DIR`.
3. you add `openstack-provision-securitygroups` to the chain `ipi-openstack-pre` (so that it always runs, and only does something if `ADDITIONAL_SECURITY_GROUP_RULES` is non-empty)
4. in the workflow, you set the environment variable.

What do you think?